src/hardware/setup.rs

```diff
@@ -374,33 +384,33 @@ pub fn setup(
 
     info!("setup internal ADCs");
 
-    let out_u_pins: OutUPins = (
-        gpioc.pc3.into_analog(),
-        gpioa.pa0.into_analog(),
-        gpioa.pa3.into_analog(),
-        gpioa.pa4.into_analog(),
-    );
-    let out_i_pins: OutIPins = (
-        gpioa.pa5.into_analog(),
-        gpioa.pa6.into_analog(),
-        gpiob.pb0.into_analog(),
-        gpiob.pb1.into_analog(),
-    );
-    let supply_pins: SupplyPins = (
-        gpioc.pc0.into_analog(),
-        gpioc.pc2.into_analog(),
-        gpiof.pf7.into_analog(),
-        gpiof.pf8.into_analog(),
-    );
+    let adc_pins = AdcPins {
+        output_voltage: (
+            gpioc.pc3.into_analog(),
+            gpioa.pa0.into_analog(),
+            gpioa.pa3.into_analog(),
+            gpioa.pa4.into_analog(),
+        ),
+        output_current: (
+            gpioa.pa5.into_analog(),
+            gpioa.pa6.into_analog(),
+            gpiob.pb0.into_analog(),
+            gpiob.pb1.into_analog(),
+        ),
+        supply: (
```

Comment:
The `supply` tuple may benefit from making it a struct with named members (because there is no natural ordering).